Tella/Data/Crypto/Cryptor.swift

```diff
@@ -0,0 +1,161 @@
+    case finalizationFailed
+}
+
+class Cryptor {
+    
+    static let bufferSize = 1024 * 1024 * 8  // 8 MB buffer size
+    
+    private var cryptor: CCCryptorRef?
+    private var cryptoOperation: CryptoOperationEnum
+    private var inputFile:FileHandle
+    private var outputFile:FileHandle
+    private var encryptionKeyData: Data
+    
+    init(inputFileURL: URL,
+         outputFileURL:URL,
+         encryptionKeyData: Data,
+         cryptoOperation: CryptoOperationEnum) throws {
+        
+        let _ = inputFileURL.startAccessingSecurityScopedResource()
+        defer { inputFileURL.stopAccessingSecurityScopedResource() }
+        
+        let _ = outputFileURL.startAccessingSecurityScopedResource()
+        defer { outputFileURL.stopAccessingSecurityScopedResource() }
+        
+        // Open the input and output file handles
+        self.inputFile = try FileHandle(forReadingFrom: inputFileURL)
+        self.outputFile = try FileHandle(forWritingTo: outputFileURL)
+        
+        self.cryptoOperation = cryptoOperation
+        self.encryptionKeyData = encryptionKeyData
+    }
+    
+    func cryptFile() throws {
+        try cryptorCreate()
+        try cryptorUpdate()
+        try cryptorFinal()
+    }
+    
+    private func cryptorCreate() throws  {
+        // Setup the encryption context
+        let keySize = size_t(kCCKeySizeAES256)
+        let algorithm = CCAlgorithm(kCCAlgorithmAES)
+        let options = CCOptions(ccNoPadding)
+        let mode = CCMode(kCCModeCTR)
+        let operation: CCOperation = cryptoOperation == .encrypt ? UInt32(kCCEncrypt) : UInt32(kCCDecrypt)
+        
+        let operationResult = encryptionKeyData.withUnsafeBytes { keyBytes in
+            
+            CCCryptorCreateWithMode(
+                operation,
+                algorithm,
+                mode,
+                options,
+                nil, // Initialization Vector
+                keyBytes.baseAddress, // Encryption Key
+                keySize,
+                nil,
+                0,
+                0,
+                CCModeOptions(kCCModeOptionCTR_BE),
+                &cryptor
+            )
+        }
+        
+        if operationResult !=  kCCSuccess {
+            debugLog("Error encryption \(operationResult)", space: .crypto)
+            throw CryptoError.cryptorCreationFailed
+        }
+    }
+    
+    private func cryptorUpdate() throws {
+        var stop = false
+        
+        // Process the file in chunks
+        while true {
+            
+            try autoreleasepool {
+                
+                // Read a chunk from the input file
+                
+                let inputBuffer = inputFile.readData(ofLength: Self.bufferSize) as Data
+                
+                var encryptedData = Data(count: Int(Self.bufferSize) + kCCBlockSizeAES128)
+                
+                if inputBuffer.isEmpty {
+                    // Reached end of file
+                    stop = true
```

Comment:
Instead of having an stop variable and checking it to exit the while loop. Why don't we just add a break statement here? something simple as this: `guard !inputBuffer.isEmpty else { break }`

If we need to execute the rest of the code for this last part of the file we can just move it to the end of the loop right?

Reply:
I think `break` doesn't work with `autoreleasepool` that's why I added a `stop` var